web/ui/react-app/src/pages/status/Status.tsx

```diff
@@ -83,18 +84,21 @@ const StatusWithStatusIndicator = withStatusIndicator(StatusContent);
 
 StatusContent.displayName = 'Status';
 
-const Status: FC = () => {
+const Status: FC<{ agentMode: boolean }> = ({ agentMode }) => {
   const pathPrefix = usePathPrefix();
   const path = `${pathPrefix}/${API_PATH}`;
 
   return (
     <>
       {[
-        { fetchResult: useFetch<Record<string, string>>(`${path}/status/runtimeinfo`), title: 'Runtime Information' },
-        { fetchResult: useFetch<Record<string, string>>(`${path}/status/buildinfo`), title: 'Build Information' },
-        { fetchResult: useFetch<Record<string, string>>(`${path}/alertmanagers`), title: 'Alertmanagers' },
-      ].map(({ fetchResult, title }) => {
-        const { response, isLoading, error } = fetchResult;
+        { FetchResult: () => useFetch<Record<string, string>>(`${path}/status/runtimeinfo`), title: 'Runtime Information' },
```

Comment:
Any reason for the capitalization of these functions? It seems unrelated to the rest of the change.

Reply:
The linter didn't like fetchResult being lowercase:
```
Failed to compile.

src/pages/status/Status.tsx
  Line 94:30:  React Hook "useFetch" is called in function "fetchResult" that is neither a React function component nor a custom React Hook function. React component names must start with an uppercase letter  react-hooks/rules-of-hooks
  Line 95:30:  React Hook "useFetch" is called in function "fetchResult" that is neither a React function component nor a custom React Hook function. React component names must start with an uppercase letter  react-hooks/rules-of-hooks
  Line 96:30:  React Hook "useFetch" is called in function "fetchResult" that is neither a React function component nor a custom React Hook function. React component names must start with an uppercase letter  react-hooks/rules-of-hooks

Search for the keywords to learn more about each error.
```

Reply:
Just capitalizing these may not be the best approach but it seemed to appease the linter. Open to suggestions here?

Reply:
Ah, some of those hook-related linter rules are really annoying, because they are not smart enough to understand enough context much of the time. In this case, the upper-casing only shuts up the linter because it now thinks that we are defining a React component, which we are not (it is actually a custom hook function, but the linter doesn't recognize it as such), so we just "tricking" the linter into believing something wrong to shut it up. Could you try renaming these to `useFetchResult` instead? Hopefully if they start with `useXXX`, the linter will detect them as custom hooks.

Reply:
in this case, the linter complains now about using a hook in a callback:
```src/pages/status/Status.tsx
  Line 101:48:  React Hook "useFetchResult" cannot be called inside a callback. React Hooks must be called in a React function component or a custom React Hook function  react-hooks/rules-of-hooks```

Reply:
Ah yeah, the general pattern here of calling hooks from a `map` callback is something that isn't super kosher, as it makes it less certain to the linter that the hooks are always all called, and in the same order (https://reactjs.org/docs/hooks-rules.html#only-call-hooks-at-the-top-level). But that was already the problem before.

If we really wanted to abide by the "rules of hooks" here, we would have to restructure everything, probably just write everything out...

Reply:
I moved this logic into a new functional component. Does this work?